ui/src/app/services/leaderboard.service.ts

```diff
@@ -1,6 +1,7 @@
 import { Injectable, OnInit } from "@angular/core";
 import { ApiService } from "./api.service";
 import { UserSessionService } from "./user-session.service";
```

Comment:
You've replaced the UserSessionService with functionality in the UserService. Remove UserSession, as it's now superfluous.